RecoPixelVertexing/PixelTriplets/src/CACut.h

```diff
@@ -0,0 +1,175 @@
+// -*- C++ -*-
+// //
+// // Package:    RecoPixelVertexing/PixelTriplets
+// // Class:      CACut
+// //
+// // Original Author:  Karla Josefina Pena Rodriguez
+// //         Created:  Wed, 14 Feb 2019 10:30:00 GMT
+// //
+
+#ifndef RecoPixelVertexing_PixelTriplets_src_CACut_h
+#define RecoPixelVertexing_PixelTriplets_src_CACut_h
+
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "CAGraph.h"
+
+class CACut {
+public:
+  explicit CACut(const double defaultCut, const std::vector<edm::ParameterSet> &tripletCuts)
+      : useCACuts(true), foundAllLayerIds(false), defaultCut_(defaultCut) {
+    if (tripletCuts.size() == 1 && tripletCuts[0].getParameter<double>("cut") == -1.) {
+      useCACuts = false;
+      //TODO Uncomment the Following warning after tuning
```

Comment:
Since there will be no tuning for Run2, this would indeed produce unnecessary noise. The idea was to have a warning when the current default cut is used instead of the tuned values e.g. in case of future changes to the configurations or CA code. Would it make sense to keep it as a LogDebug instead or should I remove it altogether?

Reply:
How about keeping it as a LogDebug instead? Or should I remove it altogether?

Reply:
And should the warning on line 71 also be a LogDebug then?

Reply:
If they need to be "warning", i.e. if they will happen rarely and they are worth being notified, they can stay as such. Otherwise you can move them to LogDebug.
The one at L71 seems really needing a warning, according to my understanding. Unless it can be fired also in "normal" situations, in which case it doesn't
In any case, staying commented out doesn't serve any purpose.

Reply:
Since there will be no tuned values for Run2, this warning would indeed produce unnecessary noise. I can move it to LogDebug, to be used when working on the tune itself

Reply:
About the warning on L71, maybe @makortel or @mtosi can comment? The CAGraph is updated for every event, and includes only layers with hits. This means the warning would also be fired for events where there was a detector layer with no hits.

Reply:
I would imagine events without hits on a pixel layer to be rare, but I would say such an event would nevertheless be a good event and not an indication of a mistake in the configuration (just my 2 cents).

Thinking more broadly, the current need to interpret the layer combinations of `SeedingLayersEDProducer` is suboptimal (and I'm guilty as charged for that). Possible ways to check the compatibility of the configurations could be to pass the layer combinations as a Run product (or eventually `ProcessBlock`), or via EventSetup. I think either would be beyond the scope of this PR.

Reply:
Thanks @makortel! Would you have any suggestions on how to improve either this particular check of the layer combinations or the warning itself? Should we make it clear that one should check the layers in the configuration, but the warning could also be fired for rare events where one of the detector layers had no hits?